Teacher-App/app/src/main/java/com/example/seeds/connectivity/ConnectivityRepository.kt

```
@@ -0,0 +1,101 @@
+    private suspend fun ping() {
+        val start = System.currentTimeMillis()
+        try {
+            withTimeout(ABORT_TIMEOUT_MS) {
+                service.healthPing()
+            }
+            val elapsed = System.currentTimeMillis() - start
+            applyStatus(
+                if (elapsed >= PING_THRESHOLD_MS) ConnectivityStatus.DEGRADED
+                else ConnectivityStatus.ONLINE
+            )
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)